distribution/docker/alpine/Dockerfile

```diff
@@ -1,20 +1,53 @@
-FROM docker.io/alpine:3.14 AS builder
+ARG ALPINE_IMAGE=docker.io/alpine:3.15
 
-WORKDIR /vector
+#-------------------------------------------------------------------------------
+FROM ${ALPINE_IMAGE} AS staging
 
-COPY vector-*-unknown-linux-musl*.tar.gz ./
-RUN tar -xvf vector-0*-"$(cat /etc/apk/arch)"-unknown-linux-musl*.tar.gz --strip-components=2
+# Set a well-known working directory for staging
+WORKDIR /staging
 
-RUN mkdir -p /var/lib/vector
+# Prepare a staging root filesystem
+RUN mkdir -p \
+  rootfs/etc/vector \
+  rootfs/usr/bin \
+  rootfs/var/lib/vector
 
-FROM docker.io/alpine:3.14
-RUN apk --no-cache add ca-certificates tzdata
+COPY docker-entrypoint.sh rootfs/usr/bin/docker-entrypoint.sh
+COPY --from=docker.io/tianon/gosu:latest gosu rootfs/usr/bin/gosu
 
-COPY --from=builder /vector/bin/* /usr/local/bin/
-COPY --from=builder /vector/config/vector.toml /etc/vector/vector.toml
-COPY --from=builder /var/lib/vector /var/lib/vector
+# Copy and unpack Vector distribution
+COPY vector-*-unknown-linux-musl*.tar.gz .
+RUN mkdir vector \
+    && tar zxvf vector-*-"$(cat /etc/apk/arch)"-unknown-linux-musl*.tar.gz \
+    -C vector --strip-components=2
 
-# Smoke test
-RUN ["vector", "--version"]
+# Prepare Vector in the staging root filesystem
+RUN cp vector/bin/* rootfs/usr/bin/ \
+  && cp -r vector/config/* rootfs/etc/vector/
 
-ENTRYPOINT ["/usr/local/bin/vector"]
+#-------------------------------------------------------------------------------
+FROM ${ALPINE_IMAGE}
+
+# Set image labels
+ARG VECTOR_VERSION
+LABEL org.opencontainers.image.source="https://github.com/vectordotdev/vector" \
+  org.opencontainers.image.title="Vector (Alpine variant)" \
+  org.opencontainers.image.url="https://vector.dev/" \
+  org.opencontainers.image.version="${VECTOR_VERSION}"
+
+# Copy root filesystem from staging
+COPY --from=staging /staging/rootfs/ /
+
+RUN apk --no-cache add \
+        ca-certificates \
+        tini \
+        tzdata \
+    && addgroup -S vector \
+    && adduser -S -H \
+        -s /sbin/nologin \
+        -G vector vector \
+    && chown vector /etc/vector /var/lib/vector
```

Comment:
I would suggest to parammetrise the user/group names with a default to `vector`, for flexibility and code robustness.
Also would recommend to explicitly set group in addition to user in `chown`, and make it recursive.

```suggestion
# Prepare a user/group for running Vector
ARG VECTOR_USER=vector
ARG VECTOR_GROUP=vector
RUN apk --no-cache add \
        ca-certificates \
        tini \
        tzdata \
    && addgroup -S "${VECTOR_GROUP}" \
    && adduser -S -H \
        -s /sbin/nologin \
        -G "${VECTOR_GROUP}" "${VECTOR_USER}" \
    && chown -r "${VECTOR_USER}":"${VECTOR_GROUP}" /etc/vector /var/lib/vector
```